syntax = "proto2";

import "header.proto";
import "config.proto";
import "command.proto";
import "objects.proto";


//---------------------------------

message GatewayStatus
{
  optional bool configured	= 1;
  optional int64 uptime	= 2;
  optional int64 timeStamp	= 3;
  optional bool online = 4;   
}
//---------------------------------

message EquipmentStatus
{
  optional bool online		= 1;
  optional int64 commUpTime	= 2;
  optional bool UNS		= 3;
  optional int64 timeStamp	= 4;
}
//---------------------------------

// Messages that will be uploaded to gateway (Server->Gateway)
message UploadToGateway
{
  required HeaderMessage header	= 1;
  optional ConfigGateway configGw = 2;
  optional Commands cmd = 3;
  optional CheckGateway checkGw = 4;
  optional EquipmentStatus eqpStatus = 5;
}

//---------------------------------
// Checks if gateway is alive
message CheckGateway
{
  required bool isAlive	= 1;
}
//---------------------------------

// Messages that will be uploaded to server (Gateway->Server)
message UploadToServer
{
  required HeaderMessage header				= 1;
  optional GatewayStatus gwStatus			= 2; // Gateway Status
  optional EquipmentStatus eqpStatus			= 3; // Equipment Status
  optional ConfigGatewayResponse configGwResponse	= 4;
  optional Commands cmd 				= 5;
}
//---------------------------------

message NotificationValues
{
  repeated ObjectValues alarmList 		= 1; // Alarms (Uns / Class Poll)
  repeated ObjectValues staticList 		= 2; // State (Class Poll)
}

// Messages that will be uploaded to server UNS (Gateway->Server UNS)
message NotificationToServer
{
  required HeaderMessage header 		= 1; // Header: required fields
  optional EquipmentStatus eqpStatus		= 2; // Equipment Status
  optional GatewayStatus gwStatus		= 3; // Gateway Status
  optional NotificationValues notificationval	= 4; //
}

//---------------------------------
